demo-circuit/src/constants/mod.rs

```diff
@@ -154,13 +154,32 @@ impl VRFParams {
     }
 }
 
+//TODO: Move these constants in cctp-lib
+pub const SC_PUBLIC_KEY_LENGTH: usize = 32;
+pub const SC_SECRET_KEY_LENGTH: usize = 32;
+pub const SC_TX_HASH_LENGTH: usize = 32;
+pub const SC_CUSTOM_HASH_LENGTH: usize = 32;
+pub const MC_RETURN_ADDRESS_BYTES: usize = 20;
+pub const FIELD_MODULUS: usize = FIELD_CAPACITY + 1;
+
+pub const MST_MERKLE_TREE_HEIGHT: usize = 22;
+
+pub const CSW_PHANTOM_PUB_KEY_BYTES: [u8; 32] = [
+    217, 127, 224, 199, 8, 45, 179, 51, 115, 161, 177, 30, 203, 183, 46, 176, 168, 185, 222, 243,
+    130, 216, 130, 102, 88, 154, 253, 135, 199, 233, 73, 48,
+];
+
 #[cfg(test)]
 mod test {
-    use super::*;
     use algebra::{AffineCurve, FpParameters, FromCompressedBits, PrimeField};
+    use cctp_primitives::utils::serialization::serialize_to_buffer;
+
+    use super::*;
     use bit_vec::BitVec;
     use blake2s_simd::{Hash, Params};
 
+    use serial_test::*;
+
```

Comment:
Not related to Line 182, but the `fn hash_to_curve()`: 
Why is not the more performant "hunting and pecking" strategy used? That one simple computes a starting x-coordinate via hash, and then increment it until we find a point for it.

Reply:
Well that's what we do it basically but instead of incrementing the x_coordinate, we increment an internal counter that produces a new hash